dev/NumberBox/NumberBox.cpp

```diff
@@ -49,6 +46,40 @@ NumberBox::NumberBox()
     SetDefaultStyleKey(this);
 }
 
+// This was largely copied from Calculator's GetRegionalSettingsAwareDecimalFormatter()
+winrt::DecimalFormatter NumberBox::GetRegionalSettingsAwareDecimalFormatter()
+{
+    std::vector<winrt::hstring> languageList;
```

Comment:
>std::vector<winrt::hstring> languageList; [](start = 4, length = 41)

Why is this a vector? seems like we only ever add one item max

Reply:
That's just what the constructor for NumberFormatter wants.

Reply:
Maybe it would be cleaner to create the vector only once you have a language name? You might be able to do it with initializer syntax in the ternary operator below, I'm not sure.

---
In reply to: [372026234](https://github.com/microsoft/microsoft-ui-xaml/pull/1894#discussion_r372026234) [](ancestors = 372026234)

Reply:
Maybe initialize the size to 0? I'm not sure if that avoids an allocation. Maybe just look at / debug through the STL implementation? https://github.com/microsoft/STL/blob/master/stl/inc/vector#L469

Reply:
I don't think I can do it in one line so I switched it around a bit.